### ISSUE TITLE
Inconsistent output dtype in mixed scalar-array operations (migrated from Trac #1827)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3375]
Original ticket http://projects.scipy.org/numpy/ticket/1827
Reported 2011-05-10 by trac user tihocan, assigned to unknown.

It looks like the resulting dtype of a "+" operation between a scalar and an array can depend on the order of the arguments. This seems wrong.

Tested with numpy 1.5.1 under Ubuntu 11.04 (64-bit).

```
In [4]: (numpy.array(0, dtype='uint16') + numpy.array([1], dtype='int8')).dtype
Out[4]: dtype('int16')

In [6]: (numpy.array([1], dtype='int8') + numpy.array(0, dtype='uint16')).dtype
Out[6]: dtype('int8')
```

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-05-10

This has been fixed in 1.6.0. Thanks for reporting.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-05-31
